tokio-util/src/io/sink_writer.rs

```diff
@@ -0,0 +1,115 @@
+use futures_sink::Sink;
+
+use pin_project_lite::pin_project;
+use std::io;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio::io::AsyncWrite;
+
+pin_project! {
+    /// Convert a [`Sink`] of byte chunks into an [`AsyncWrite`].
+    ///
+    /// Each write to the [`SinkWriter`] is converted into a send of `&[u8]` to the [`Sink`].
+    /// Flushes and shutdown are propagated to the sink's flush and close methods.
```

Comment:
This sentence is confusing.
```suggestion
    /// Whenever you write to this [`SinkWriter`], the supplied bytes are forwarded to the
    /// inner [`Sink`]. When [`shutdown`] is called on this [`SinkWriter`], the inner sink is closed.
```

Reply:
I think for clarity we should mention the copy there, otherwise not married to that formulation.

Reply:
There's no copy here - `SinkWriter` does no copying of its own, it just forwards bytes to the `Sink`.